rohmu/object_storage/azure.py

```diff
@@ -113,6 +113,35 @@ def __init__(
         self.container = self.get_or_create_container(self.container_name)
         self.log.debug("AzureTransfer initialized, %r", self.container_name)
 
+    @staticmethod
+    def conn_string(
+        account_name: str,
+        account_key: Optional[str],
+        azure_cloud: Optional[str],
+        host: Optional[str],
+        port: Optional[int],
+        is_secure: bool,
+    ) -> str:
+        protocol = "https" if is_secure else "http"
+        conn = [
+            f"DefaultEndpointsProtocol={protocol}",
+            f"AccountName={account_name}",
+            f"AccountKey={account_key}",
+        ]
+        if not host and not port:
+            try:
+                endpoint_suffix = ENDPOINT_SUFFIXES[azure_cloud]
+            except KeyError:
+                raise InvalidConfigurationError(f"Unknown azure cloud {repr(azure_cloud)}")
+
+            conn.append(f"EndpointSuffix={endpoint_suffix}")
+        else:
+            if not host or not port:
+                raise InvalidConfigurationError("Custom host and port must be specified together")
```

Comment:
It would be better to use pydantic validators for custom validation, see https://docs.pydantic.dev/1.10/usage/validators/

Reply:
Good idea. Adding this on the latest commit, have a look

Reply:
The validation of `azure_cloud` can also be moved in the validator. Also, remember to fold fixup commits before pushing :)

Reply:
Sure, can move it to validators as well.

Re: fixups, I prefer to keep them separate for easier review, but can surely rebase/squash once approved if that's ok.

Reply:
That is also fine, and I used to recommend that too before. I just find it unnecessary given that GitHub provides the diff between the previous and current branch tip on force push (via the "Compare" button).

Reply:
Also if a reviewer marks the files as "viewed" when reviewing GitHub will also show them which files have changed when the PR is updated, so it's not like reviewers are forced to re-read everything to discover what you updated.

Reply:
Thanks, will keep that in mind. 
I've squashed all commits now for this change.